trl/trainer/grpo_trainer.py

```diff
@@ -48,10 +49,14 @@
 
 
 if is_peft_available():
-    from peft import PeftConfig, PeftModel, get_peft_model
+    from peft import LoraConfig, PeftConfig, PeftModel, get_peft_model
 
 if is_vllm_available():
     from vllm import LLM, SamplingParams
+    from vllm.lora.request import LoRARequest
+
+    # Enable runtime LoRA updating on every new checkpoint
+    os.environ["VLLM_ALLOW_RUNTIME_LORA_UPDATING"] = "True"
```

Comment:
I'm not a fan of permanently setting env vars, as this can lead to hard to debug errors. Ideally, something like [`patch_environment`](https://github.com/huggingface/accelerate/blob/ce63623421750c95276971ba1086e89dcdc0d112/src/accelerate/utils/environment.py#L314) from accelerate could be used instead. Otherwise, a destructor (`__del__`) could be defined on the appropriate object but that sounds brittle.